### PR TITLE
[reliability] Consistent handling of direction_id when NaNs present

### PR DESCRIPTION
**Pull request by yiyange**
Fix issue https://github.com/kuanb/peartree/issues/89

### CODE REVIEW

File: peartree/parallel.py [hunk 1]
```diff
@@ -165,7 +175,7 @@ def generate_all_observed_edge_costs(trips_and_stop_times: pd.DataFrame
                 dir_mask = (tst_sub.direction_id == direction)
                 tst_sub_dir = tst_sub[dir_mask]
             else:
-                tst_sub_dir = tst_sub
+                tst_sub_dir = tst_sub.copy()
```
Comment (kuanb) [inline peartree/parallel.py]: Why the removal of the `.copy()`? Without examining it deeply, shouldn't we defensively employ `.copy()` to avoid upstream impacts? It's removal does not seem to be relevant to handling the `direction_id` column being dropped.

Comment (kuanb) [reply peartree/parallel.py]: Ignore this. I am so sorry.

File: peartree/parallel.py [hunk 2]
```diff
@@ -74,9 +74,19 @@ def generate_route_costs(self, route_id: str):
                      'departure_time']
         trips_and_stop_times = trips_and_stop_times.sort_values(sort_list)
 
+        # Check direction_id column value before
```
Comment (kuanb) [inline peartree/parallel.py]: Per our discussion offline - the advantage to adding the handling here is that it is within a single route. This means that we do not end up tossing direction id if a specific route happens to have all direction id rows filled in. It is possible that a route operator has direction id for one route but not another.

Comment (yiyange) [reply peartree/parallel.py]: comments added. thanks!

**Comment by kuanb** [comment]
@yiyange made some slight changes to the comments (see above commit).

**Comment by kuanb** [comment]
Final request, please add a test in `test_paths.py`. Something along the lines of:
```
def test_loading_in_too_small_timeframes():
    path = fixture('samtrans-2017-11-28.zip')
    feed = get_representative_feed(path)

    # Overwrite the stop times so that direction id is null
    st = feed.stop_times
    st.direction_id = np.nan
    feed.stop_times = st

    # Make sure the behavior is as expected
    start = 7 * 60 * 60
    end = 10 * 60 * 60
    G = load_feed_as_graph(feed, start, end)

    # Add an assertion that stops times are not null
```

File: peartree/parallel.py [hunk 3]
```diff
@@ -74,9 +74,21 @@ def generate_route_costs(self, route_id: str):
                      'departure_time']
         trips_and_stop_times = trips_and_stop_times.sort_values(sort_list)
 
+        # Check direction_id column value before using
+        # trips_and_stop_times to generate wait and edge costs
+        # Note: Advantage to adding handling at route level is that peartree
+        #       avoids tossing direction id if a specific route has all direction
+        #       id rows filled in (while another does not, which is possible).
```
Comment (yiyange) [inline peartree/parallel.py]: evidence for the aforementioned case. `this is happening` logs where there are non values in direction_id column. it is clear that some routes have full coverage but some do not. therefore we can't move this check further upstream.
```
	Reduced selected trips on route 295-155 from 12 to 9.
this is happening
	Reduced selected trips on route 296-155 from 45 to 29.
this is happening
	Reduced selected trips on route 297-157 from 8 to 0.
	Reduced selected trips on route 397-157 from 7 to 0.
	Reduced selected trips on route 398-157 from 11 to 6.
this is happening
	Reduced selected trips on route 399-157 from 9 to 0.
	Reduced selected trips on route ECR-155 from 59 to 41.
```
